messages/src/lib.rs

```diff
@@ -41,6 +43,64 @@ pub enum Error {
     UnsupportedAlgorithmIdentifier(&'static str, u16),
 }
 
+/// ASCII-encoded URL with minimum length 1 and maximum length 2^16.
+#[derive(Clone, PartialEq, Eq)]
+pub struct Url(Vec<u8>);
```

Comment:
This struct is defined by DAP, hence why I put it in the parent mod. https://www.ietf.org/archive/id/draft-ietf-ppm-dap-05.html#name-protocol-definition